src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java

```diff
@@ -22,16 +23,20 @@ public ExtendAndScoreCone(DrivetrainSubsystem drive, FloorArmSubsystem floorArm,
       new WaitCommand(0.5),
       new FloorMoveArmToPostion(floorArm, Constants.FLOOR_ARM_INSIDE_ROBOT_POSITION),
       new MoveArmToPosition(arm, Constants.ARM_SCORE_HIGH_POSITION),
-      //new WaitUntilCommand(() -> Math.abs(arm.getPosition() - arm.getTargetPosition()) < Constants.ARM_ENCODER_COUNT_ERROR),
       new WaitCommand(1),
-      new SetDriveTranslationTarget(drive, getTranslationTarget(drive, 1)),
-      new DropCone(intake).withTimeout(1),//could probably tune this down
-      new SetDriveTranslationTarget(drive, getTranslationTarget(drive, -1)),
+      driveDistance(drive, 0.5, -1),
+      new DropCone(intake).withTimeout(0.5),
+      driveDistance(drive, 0.5, 1).withTimeout(0.5),
       new MoveArmToPosition(arm, Constants.ARM_INSIDE_ROBOT_POSITION)
     );
   }
 
-  private Translation2d getTranslationTarget(DrivetrainSubsystem drive, double dx) {
-    return drive.getPose().getTranslation().plus(new Translation2d(dx, 0));
+  private Command driveDistance(DrivetrainSubsystem drive, double distance, double velocity) {
+    return new StartEndCommand(
+      () -> new RepeatCommand(new InstantCommand(() ->drive.drive(velocity, 0, 0), drive)),
```

Comment:
This should be able to just be a drive method call instead of needing the repeat command. Since the chassis speed persist this command will set them when it’s initialized and set them to 0 when it ends. Since the speeds are constant we don’t need to mimic the execute method with a repeat command